src/firecrest/dependencies.py

```diff
@@ -356,3 +359,72 @@ def __eq__(self, other):
     # To allow for dependency override hash is based on class
     def __hash__(self):
         return hash(S3ClientDependency.__class__)
+
+
+class DataTransferDependency:
+    def __init__(self):
+        pass
+
+    # Note: this fuction allows for unit test client injection override
+    async def _get_ssh_client(self, system_name):
+        return await SSHClientDependency(ignore_health=False)(system_name=system_name)
+
+    async def _get_scheduler_client(self, system_name):
+        return await SchedulerClientDependency()(system_name=system_name)
+
+    def _get_s3_client(self, endpoint_url):
+        return get_session().create_client(
+            "s3",
+            region_name=settings.storage.region,
+            aws_secret_access_key=settings.storage.secret_access_key.get_secret_value(),
+            aws_access_key_id=settings.storage.access_key_id.get_secret_value(),
+            endpoint_url=endpoint_url,
+            config=AioConfig(signature_version="s3v4"),
+        )
+
+    async def __call__(
+        self,
+        system_name: str,
+    ):
+        system = ServiceAvailabilityDependency(service_type=HealthCheckType.s3)(
```

Comment:
A comment for future implementations: we should create an abstraction for Data Transfers health checks. In this case, this works because the only specialization at the time is with `S3DataTransfer`, but when we start adapting other classes without support for S3 (Globus, Direct Transfers, etc) this won't adapt to the health check.

Reply:
Yes, I left this part as is for now. This should be changed once we introduce a new transfer type.